plugins/src/lib.rs

```diff
@@ -167,42 +150,45 @@ impl Parse for DeriveSerde {
 pub fn service(attr: TokenStream, input: TokenStream) -> TokenStream {
     let derive_serde = parse_macro_input!(attr as DeriveSerde);
 
+    let empty_tuple = quote!(());
+
     let Service {
         attrs,
         vis,
         ident,
         rpcs,
     } = parse_macro_input!(input as Service);
-    let vis_repeated = std::iter::repeat(vis.clone());
 
-    let camel_case_fn_names: Vec<String> = rpcs
+    let camel_case_fn_names: &Vec<String> = &rpcs
         .iter()
         .map(|rpc| snake_to_camel(&rpc.ident.to_string()))
         .collect();
-    let outputs: &Vec<TokenStream2> = &rpcs
+    let outputs: &Vec<Option<TokenStream2>> = &rpcs
         .iter()
         .map(|rpc| match rpc.output {
-            ReturnType::Type(_, ref ty) => quote!(#ty),
-            ReturnType::Default => quote!(()),
+            ReturnType::Type(_, ref ty) => Some(quote!(#ty)),
+            ReturnType::Default => None,
```

Comment:
The problem here is that we need to construct new `TokenStream` in case of `Return::Type` and don't need to create new TokenStream in case of `ReturnType::Default`. So if we return `&empty_tuple`, branches will have incompatible types. Workaround here is to collect `Option<TokenStream>` and after all take collection by ref, replacing `None` by `&TokenStream`.
Another option is to clone `empty_tuple` in every `None` but it will create unnecessary `TokenStream`.

Reply:
> Could this be `&empty_tuple`, avoiding the second collect?

Solved by using `&Type` instead of `TokenStream`.